### PR TITLE
Added RM summary section & tab extensibility examples

### PR DESCRIPTION
**Pull request by anangaur**

### CODE REVIEW

File: rm-deployment-status-enhancer-summary/create.bat [hunk 1]
```diff
@@ -0,0 +1 @@
+tfx extension create --publisher dhaundiyal-railseatalert --manifest-globs vss-extension.json
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/create.bat]: Do you want this publisher name hard-coded like this in the file?

File: rm-deployment-status-enhancer-summary/bower.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/bower.json]: Can you come up with a shorter name for this folder? "rm-deployment-status-enhancer-summary" is really long. One option is to have a "release-management" root folder, that may contain one or more folders with different samples. It a more sustainable model ... and one we will start moving towards as we get more samples.

File: rm-deployment-status-enhancer-summary/index.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+ <html xmlns="http://www.w3.org/1999/xhtml">
+ <head>
+     <title>Deployment status</title>
+     <script src="sdk/scripts/VSS.SDK.js"></script>
+     <script type="text/javascript">VSS.init({ usePlatformScripts: true });</script>
+	 <script src="scripts/main.js"></script>
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/index.html]: Formatting issues with this line

File: rm-deployment-status-enhancer-summary/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			
+			container.append(p1);
+			container.append(p2);
+			$('#environments').append(container);
+		});
+	});
+});
+
+ VSS.register("sample-id", function (param) {
+	 alert(param);
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: Remove alert

File: rm-deployment-status-enhancer-summary/vss-extension.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	"id": "rm-deployment-status-enhancer-summary",
+	"version": "0.1.5",
+	"name": "RM deployment status - summary",
+	"description": "RM summary section extension sample",
+	"publisher": "openalm",
+	"links": {
+        "learn": {
+            "uri": "https://github.com/Microsoft/vso-extension-samples"
+        }
+    },
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: formatting

File: rm-deployment-status-enhancer-summary/vss-extension.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	"targets": 
+	[
+		{
+			"id": "Microsoft.VisualStudio.Services"
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: Works on-prem (update 2), too? If not, use "Microsoft.VisualStudio.Services.Cloud" instead

File: rm-deployment-status-enhancer-summary/vss-extension.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	],
+	"files": 
+	[
+		{
+			"path": "index.html", "addressable": true
+		},
+		{
+			"path": "scripts", "addressable": true
+		},     
+        {
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: Formatting

File: rm-deployment-status-enhancer-tab/create.bat [hunk 8]
```diff
@@ -0,0 +1 @@
+tfx extension create --publisher dhaundiyal-railseatalert --manifest-globs vss-extension.json
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-tab/create.bat]: same comment about publisher

File: rm-deployment-status-enhancer-summary/scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+	return {
+		// Called when the active work item is modified
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: active work item??

File: rm-deployment-status-enhancer-summary/scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
+
+VSS.ready(function() {
+	var c = VSS.getConfiguration();
+	c.onReleaseChanged(function(release) {
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: Why aren't these functions attached via registration of an object via VSS.register ?

File: rm-deployment-status-enhancer-tab/bower.json [hunk 11]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (willsmythe) [inline rm-deployment-status-enhancer-tab/bower.json]: I would combine this with the previous sample ... it's more realistic and avoids a lot of duplication.

File: rm-deployment-status-enhancer-summary/create.bat (hunk 1 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/create.bat]: Removed create.bat as it is not required

File: rm-deployment-status-enhancer-summary/bower.json (hunk 2 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/bower.json]: Done.

File: rm-deployment-status-enhancer-summary/index.html (hunk 3 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/index.html]: Formatted it

File: rm-deployment-status-enhancer-summary/scripts/main.js (hunk 4 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: VSS.register us not required.

File: rm-deployment-status-enhancer-summary/vss-extension.json (hunk 5 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: Formatted it

File: rm-deployment-status-enhancer-summary/vss-extension.json (hunk 7 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: Formatted it

File: rm-deployment-status-enhancer-tab/create.bat (hunk 8 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-tab/create.bat]: removed the unnecessary file

File: rm-deployment-status-enhancer-summary/scripts/main.js (hunk 9 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: the function is not required. It was a test fn

File: rm-deployment-status-enhancer-summary/vss-extension.json (hunk 6 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/vss-extension.json]: Should work in on-prem TFS too.

File: rm-deployment-status-enhancer-summary/scripts/main.js (hunk 10 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-summary/scripts/main.js]: It seems there is an issue using this fn wherein one cannot call into VSTS/RM service from the extension. I will get more info on this and if possible will use the fn. We can keep this change for the next iteration.

File: rm-deployment-status-enhancer-tab/bower.json (hunk 11 above)
Comment (agauravgit) [inline rm-deployment-status-enhancer-tab/bower.json]: Done

**Comment by willsmythe** [comment]
Looks good. Thanks for adding, Anand.

**Pull request merged**